Feature/tdi 35766

chmyga opened the pull request:
Added initial implementation of tSplunkEventCollector component. Unit and integration tests are still to be added. Any comments are welcome.

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/TSplunkEventCollectorProperties.java:
```diff
@@ -0,0 +1,120 @@
+        FieldAssembler<Schema> record = SchemaBuilder.record("Main").fields();
+        for(SplunkJSONEventField metadataField : SplunkJSONEventField.getMetadataFields()) {
+            record.name(metadataField.getName())
+            .prop(Talend6SchemaConstants.TALEND6_COLUMN_IS_NULLABLE, "true")
```
Don't use anything in Talend6SchemaConstants - that will go away and is DI specific. For the nullable stuff, use the code in AvroUtils.

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/connection/TSplunkEventCollectorConnection.java:
```diff
@@ -0,0 +1,78 @@
+
+/**
+ * created by dmytro.chmyga on Apr 26, 2016
+ */
```
Please remove these "created by..." headers - we don't use those.

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/TSplunkEventCollectorProperties.java:
```diff
@@ -0,0 +1,120 @@
+    private void setupDefaultSchema() {
+        AvroRegistry avroReg = new AvroRegistry();
+        FieldAssembler<Schema> record = SchemaBuilder.record("Main").fields();
+        for(SplunkJSONEventField metadataField : SplunkJSONEventField.getMetadataFields()) {
```
I think the formatting standard that you have is not correct. Can you get the correct one from @sgandon? We have a space after "for" for example.

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/runtime/TSplunkEventCollectorSink.java:
```diff
@@ -0,0 +1,97 @@
+        if(serverUrl == null || serverUrl.trim().isEmpty()) {
+            LOGGER.debug("Server URL is empty.");
+            result.setStatus(Result.ERROR).setMessage("Server URL cannot be empty.");
+        }
```
The messages need to be internationalized (the setMessage, but not the logger).

francisu commented:
Very nice work Dmytro! I made a few comments that should be corrected before we merge this.

gonchar-ivan commented on components-splunk/src/main/java/org/talend/components/splunk/TSplunkEventCollectorProperties.java:
```diff
@@ -0,0 +1,120 @@
+    public Property extendedOutput = PropertyFactory.newBoolean("extendedOutput");
+    public Property RESPONSE_CODE;
+    public Property ERROR_MESSAGE;
+    protected transient PropertyPathConnector MAIN_CONNECTOR = new PropertyPathConnector(Connector.MAIN_NAME, "schema");
```
Why is it transient?
I haven't find this Properties implements Serializable.

gonchar-ivan commented on components-splunk/src/main/java/org/talend/components/splunk/connection/TSplunkEventCollectorConnection.java:
```diff
@@ -0,0 +1,78 @@
+        if(client == null) {
+            try {
+                client = new DefaultHttpClient();
+            } catch (Exception e) { 
```
It is no good to leave empty catch clause
add logger.debug(e.getMessage()); maybe

gonchar-ivan commented on components-splunk/src/main/java/org/talend/components/splunk/objects/SplunkJSONEvent.java:
```diff
@@ -0,0 +1,60 @@
+        jsonEventObject.put(key, object);
+    }
+    
+    public void put(String name, Object value) {
```
add @Override maybe?

chmyga commented:
Fixed most of the problems found during the first review:
-removed Talend6SchemaConstants usage;
-remove "created by..." headers;
-formatted the code;
-added internationalization for messages;
-removed unneeded try-catch block;
-added some exceptions to runtime for sending failures;
-fixed validation in Sink;
-added some tests, some more are to be added in the nearest future.

Please, review the code if you have some time.

sgandon commented on components-splunk/src/main/java/org/talend/components/splunk/connection/TSplunkEventCollectorConnection.java:
```diff
@@ -14,29 +14,30 @@
  */
 public class TSplunkEventCollectorConnection {
 
+    private transient static final Logger LOGGER = LoggerFactory.getLogger(TSplunkEventCollectorConnection.class);
```
Great job of adding all these logging messages.

sgandon commented on components-splunk/src/main/java/org/talend/components/splunk/runtime/TSplunkEventCollectorWriter.java:
```diff
@@ -102,40 +107,40 @@ public void write(Object datum) throws IOException {
+            lastErrorCode = ((Long) json.get("code")).intValue();
+            lastErrorMessage = (String) json.get("text");
         } catch (ParseException e) {
             e.printStackTrace();
```
no e.printStackTrace() please.
You should handle this exception to that is surface to the calling method, I would suggest you create an IOException and throw it.

sgandon commented on components-splunk/src/test/java/org/talend/components/splunk/TSplunkEventCollectorTestIT.java:
```diff
@@ -0,0 +1,92 @@
+
+    @Test
+    public void testGetProps() {
+        ComponentProperties props = new TSplunkEventCollectorDefinition().createProperties();
```
I wonder what is this test for ? because serialization and i18n is already tested in the framework.

sgandon commented on components-splunk/src/test/java/org/talend/components/splunk/TSplunkEventCollectorTestBase.java:
```diff
@@ -0,0 +1,56 @@
+import org.talend.daikon.properties.presentation.Form;
+
+public class TSplunkEventCollectorTestBase extends AbstractComponentTest {
+
```
You don't need to extends AbstractComponentTest and implement getComponentService if you are using SpringJUnit4ClassRunner. Spring shall find your component upon starting scan. You only need this is you don't use Spring.

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/TSplunkEventCollectorProperties.java:
```diff
@@ -0,0 +1,126 @@
+ * <li>Validating the properties of the component at design-time,</li>
+ * <li>Containing the untyped values of the properties, and</li>
+ * <li>All of the UI information for laying out and presenting the properties to the user.</li>
+ * </ol>
```
You don't need these comments on lines 25-31 as they are generic. I'm sorry for being so picky, but I really don't want to have any code or comments that we don't need anywhere.  They belong in the ComponentProperties class (but I think that already has good Javadoc). If you think it should be improved, can you make a pull request for that?

francisu commented on components-splunk/src/main/java/org/talend/components/splunk/TSplunkEventCollectorProperties.java:
```diff
@@ -0,0 +1,126 @@
+package org.talend.components.splunk;
+
```
Need a license file.

chmyga replied:
Yes, I've forgotten to add those headers to the files, generated by using mvn archetype. Maybe we should add those to the archetype?

sgandon replied:
Yes definitely, please add those headers in the Archetype files.

francisu commented:
I'm going to merge this, and can you handle the remaining issues directly? Again, nice work.
